### PR TITLE
iCEM fixes

### PR DESCRIPTION
**Pull request by luisenp**
## Types of changes
Opened this pull request to suggest some changes to `icem` branch. This also rebases `icem` branch onto `master`, which is why I didn't try to commit to that branch directly (and why merge has conflicts). I would nevertheless prefer to merge this branch into the main repo to keep a simpler commit history. 

I tested the code on gym's HalfCheetah and Hopper using the `control_env.py` script in the `diagnostics` folder. Performance seems very similar to regular CEM, with slightly smaller running time and total reward. I haven't looked at the code for `powerlaw_psd_gaussian` yet. Have you tested this in some way?

- [x] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] The documentation is up-to-date with the changes I made.
- [x] I have read the [**CONTRIBUTING**](../../CONTRIBUTING.md) document and completed the CLA (see **CONTRIBUTING**).
- [x] All tests passed, and additional code has been covered with new tests.

### CODE REVIEW

File: mbrl/planning/trajectory_opt.py [hunk 1]
```diff
@@ -153,6 +153,327 @@ def optimize(
...
+        population_decay_factor: float,
+        colored_noise_exponent: float,
+        lower_bound: Sequence[Sequence[float]],
+        upper_bound: Sequence[Sequence[float]],
+        keep_elite_frac: float,
+        alpha: float,
+        device: torch.device,
+        return_mean_elites: bool = False,
+        round_population: bool = False,
+        ensemble_size: int = None,
```
Comment (luisenp) [inline mbrl/planning/trajectory_opt.py:323]: It would be better to remove `ensemble_size` and `round_population` and replace them with a single kwarg like `population_size_modulo: Optional[int] = None`, which does nothing if it's `None`, otherwise does the rounding. There is probably a better name for this than the one I'm suggesting, but hopefully the idea is clear.

File: mbrl/planning/trajectory_opt.py [hunk 2]
```diff
@@ -153,6 +153,327 @@ def optimize(
...
+                np.max(
+                    (
+                        self.population_size * self.population_decay_factor ** -i,
+                        2 * self.elite_num,
+                    )
+                )
+            ).astype(np.int32)
+
+            # to ensure population size is a multiple of the number of models in the ensemble
+            if self.round_population is True:
```
Comment (luisenp) [inline mbrl/planning/trajectory_opt.py:386]: I haven't looked at this carefully, but my guess is that this logic can probably be simplified. I'll try to make a more concrete suggestion later.

File: mbrl/planning/trajectory_opt.py [hunk 3]
```diff
@@ -153,6 +153,327 @@ def optimize(
...
+                self.colored_noise_exponent,
+                size=(decay_population_size, x0.shape[1], x0.shape[0]),
+                device=self.device,
+            ).transpose(1, 2)
+            population = torch.clamp(
+                population * torch.sqrt(var) + mu, self.lower_bound, self.upper_bound
+            )
+
+            if self.elite is not None:
+                keep_elites = torch.index_select(
```
Comment (luisenp) [inline mbrl/planning/trajectory_opt.py:430]: Maybe better as `kept_elites`?

File: mbrl/planning/trajectory_opt.py [hunk 4]
```diff
@@ -153,6 +153,327 @@ def optimize(
...
+                if i == 0:
+                    end_action = (
+                        torch.normal(
+                            mu[-1, :].repeat(keep_elites.shape[0], 1),
+                            torch.sqrt(var[-1, :]).repeat(keep_elites.shape[0], 1),
+                        )
+                        .unsqueeze(1)
+                        .to(self.device)
+                    )
+                    keep_elites_shift = torch.cat(
```
Comment (luisenp) [inline mbrl/planning/trajectory_opt.py:446]: Maybe better as `kept_elites_shifted`?

File: mbrl/util/math.py [hunk 5]
```diff
@@ -299,3 +301,89 @@ def propagate(
...
+        size = [size]
+    else:
+        size = list(size)
+
+    # The number of samples in each time series
+    samples = size[-1]
+
+    # Calculate Frequencies (we assume a sample rate of one)
+    # Use fft functions for real output (-> hermitian spectrum)
+    f = rfftfreq(samples, device=device)
```
Comment (luisenp) [inline mbrl/util/math.py:347]: This uses `torch>=1.8`, which is not the version tested in the release repo, so some users might not be at this version yet.  Can we change this for a local wrapper that checks `torch`'s version and if less than 1.8 uses `numpy.fft.rfftfreq` instead?

**Comment by luisenp** [comment]
Hi @sradicwebster, any comments on this?

**Comment by sradicwebster** [comment]
Hi @luisenp. Thanks for reviewing, I have been very busy so not had chance to look at this yet, will try to fit it in next week
